File: PurchaseSale/Web/configs/entities/Sale.js
module.exports = {
    Name: "Sale",
    PrimaryKey: "SaleId",
    Properties: GetProperties(),
    SaleTypeDataSource: GetSaleTypeDataSource(),
    ProductDataSource: GetProductDataSource(),
    UserDataSource: GetUserDataSource(),
    SaleStatusDataSource: GetSaleStatusDataSource(),
    AmountTypeDataSource: GetAmountTypeDataSource()
}

function GetProperties() {
    return [
        GetProperty("SaleId", "SaleId"),
        GetProperty("SaleCode", "销售单号"),
        GetProperty("SaleDate", "销售日期"),
        GetProperty("SaleUserName", "销售员"),
        GetProperty("LogisticsFee", "物流费"),
        GetProperty("OtherFee", "其他费"),
        GetProperty("DiscountFee", "折扣费"),
        GetProperty("ShouldAmount2", "应收金额"),
        GetProperty("RealAmount2", "实收金额"),
        GetProperty("CustomerName", "顾客姓名"),
        GetProperty("CustomerPhone", "顾客手机"),
        GetProperty("SaleTypeName", "类型"),
        GetProperty("SaleStatusName", "状态"),
        GetProperty("SaleAmount", "商品金额"),
        GetProperty("BidAmount", "商品成本"),
        GetProperty("Profit", "商品利润"),
        GetProperty("ProfitRate", "利润率(%)"),
        GetProperty("AmountType", "收款状态"),
        GetProperty("DueAmount", "待收金额"),
        GetProperty("Remark", "备注"),
        GetProperty("CreateDate", "创建时间")
    ]
}

function GetProperty(Name, Label) { return { Name, Label } }

function GetSaleTypeDataSource() {
    return [{ Value: 1, Text: "销售" }, { Value: 2, Text: "退货" }]
}

function GetSaleStatusDataSource() {
    return [{ Value: 0, Text: "待提交" }, { Value: 1, Text: "已提交" }, { Value: 2, Text: "已存档" }, { Value: 3, Text: "已作废" }]
}

function GetAmountTypeDataSource() {
    return [{ Value: "未结款", Text: "未结款" }, { Value: "部分结款", Text: "部分结款" }, { Value: "已结清", Text: "已结清" }]
}

function GetProductDataSource() {
    return {
        ValueName: "Id",
        TextName: "ProductName",
        StateName: "Products",
        ServiceName: "SaleService",
        ActionName: "GetProducts",
        IsRefresh: true,
        Payload: {}
    }
}

function GetUserDataSource() {
    return {
        ValueName: "UserId",
        TextName: "UserName",
        StateName: "Users",
        ServiceName: "StockCheckService",
        ActionName: "GetUsers",
        IsRefresh: true,
        Payload: {}
    }
}

File: PurchaseSale/Api/PurchaseSale.Entity/Sale.cs
using OpenDataAccess.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PurchaseSale.Entity
{
    [TableProperty(Name = "t_Sale", PrimaryKey = "SaleId", NoSelectNames = "IsDelete")]
    [RequestMethod(IsDelete = false)]
    public class Sale : EntityModel, IEntity
    {
        /// <summary> 
        /// 主键
        /// </summary> 
        public Guid SaleId { get; set; }
        /// <summary>
        /// 销售单号
        /// </summary>
        public string SaleCode { get; set; }
        /// <summary>
        /// 销售单号
        /// </summary>
        public int SaleIntCode { get; set; }
        /// <summary>
        /// 销售人
        /// </summary>
        public Guid SaleUser { get; set; }
        /// <summary>
        /// 销售日期
        /// </summary>
        public DateTime SaleDate { get; set; }
        /// <summary>
        /// 销售类型，1：出货，2：退货
        /// </summary>
        public byte SaleType { get; set; }
        /// <summary>
        /// 物流费
        /// </summary>
        public decimal LogisticsFee { get; set; }
        /// <summary>
        /// 其他费
        /// </summary>
        public decimal OtherFee { get; set; }
        /// <summary>
        /// 折扣费
        /// </summary>
        public decimal DiscountFee { get; set; }
        /// <summary>
        /// 应收金额
        /// </summary>
        public decimal ShouldAmount { get; set; }
        /// <summary>
        /// 实收金额
        /// </summary>
        public decimal RealAmount { get; set; }
        /// <summary>
        /// 顾客姓名
        /// </summary>
        public string CustomerName { get; set; }
        /// <summary>
        /// 顾客手机
        /// </summary>
        public string CustomerPhone { get; set; }
        /// <summary>
        /// 销售状态,0:保存,1:提交,2:存档，3：作废            
        /// </summary>
        public byte SaleStatus { get; set; }
        /// <summary>
        /// 创建人
        /// </summary>
        public Guid CreateUser { get; set; }
        /// <summary>
        /// 更新人
        /// </summary>
        public Guid UpdateUser { get; set; }
        /// <summary>
        /// 更新时间
        /// </summary>
        public DateTime UpdateDate { get; set; }
        /// <summary> 
        /// 备注
        /// </summary> 
        public string Remark { get; set; }
        /// <summary> 
        /// 逻辑删除，1：删除,0:正常
        /// </summary> 
        public byte IsDelete { get; set; }
        /// <summary> 
        /// 创建时间
        /// </summary> 
        public DateTime CreateDate { get; set; }

        public string RowVersion { get; set; }
    }

    [TableProperty(Name = "v_Sale", PrimaryKey = "SaleId", NoSelectNames = "IsDelete")]
    [RequestMethod(IsDelete = false, IsPost = false, IsPut = false)]
    public class ViewSale : Sale
    {
        public string SaleUserName { get; set; }
        public string SaleTypeName { get; set; }
        public string SaleStatusName { get; set; }
        public decimal ShouldAmount2 { get; set; }
        public decimal RealAmount2 { get; set; }
        public decimal BidAmount { get; set; }
        public decimal SaleAmount { get; set; }
        public decimal Profit { get; set; }
        public decimal ProfitRate { get; set; }
        public string AmountType { get; set; }
    }
}
